test/Unit/Task/ESLintTest.php

```diff
@@ -0,0 +1,211 @@
+                'format' => null,
+                'max_warnings' => null,
+                'no_eslintrc' => false,
+                'quiet' => false,
+            ]
+        ];
+    }
+
+    public function provideRunContexts(): iterable
+    {
+        yield 'run-context' => [
+            true,
+            $this->mockContext(RunContext::class)
+        ];
+
+        yield 'pre-commit-context' => [
+            true,
+            $this->mockContext(GitPreCommitContext::class)
+        ];
+
+        yield 'other' => [
+            false,
+            $this->mockContext()
+        ];
+    }
+
+    public function provideFailsOnStuff(): iterable
+    {
+        yield 'exitCode1' => [
+            [],
+            $this->mockContext(RunContext::class, ['hello.js']),
+            function () {
+                $this->mockProcessBuilder('eslint', $process = $this->mockProcess(1));
+
+                $this->formatter->format($process)->willReturn($message = 'message');
+                $this->formatter->formatErrorMessage([$message], Argument::any())->willReturn('nope');
+            },
+            'nope',
+            FixableTaskResult::class
+        ];
+    }
+
+    public function providePassesOnStuff(): iterable
+    {
+        yield 'exitCode0' => [
+            [],
+            $this->mockContext(RunContext::class, ['hello.js']),
+            function () {
+                $this->mockProcessBuilder('eslint', $this->mockProcess(0));
+            }
+        ];
+    }
+
+    public function provideSkipsOnStuff(): iterable
+    {
+        yield 'no-files' => [
+            [],
+            $this->mockContext(RunContext::class),
+            function () {}
+        ];
+        yield 'no-files-after-triggered-by' => [
+            [],
+            $this->mockContext(RunContext::class, ['notajsfile.txt']),
+            function () {}
+        ];
+        yield 'no-files-after-whitelist-patterns' => [
+            [
+                'whitelist_patterns' => ['/^resources\/js\/(.*)/'],
+            ],
+            $this->mockContext(RunContext::class, ['resources/dont/find/this/file.js']),
+            function () {}
+        ];
+    }
+
+    public function provideExternalTaskRuns(): iterable
+    {
+        yield 'bin' => [
+            [
+                'bin' => 'node_modules/.bin/eslint',
+            ],
+            $this->mockContext(RunContext::class, ['hello.js', 'hello2.js']),
+            'eslint',
+            [
+                'node_modules/.bin/eslint',
+                'hello.js',
+                'hello2.js',
+            ]
+        ];
+        yield 'config' => [
+            [
+                'config' => '.eslintrc.json',
+            ],
+            $this->mockContext(RunContext::class, ['hello.js', 'hello2.js']),
+            'eslint',
+            [
+                '--config=.eslintrc.json',
+                'hello.js',
+                'hello2.js',
+            ]
+        ];
+        yield 'config' => [
```

Comment:
You named all keys "config" from this point on

Reply:
oops, fixed ✅